### PR TITLE
Tt 148

### PR DESCRIPTION
**Pull request by elijah-parrish**
Should be all good. Some comments and things left in for other tickets (like authorization and dealing with status and priority of charges)

### CODE REVIEW

File: src/components/ChargeAdmin.vue [hunk 1]
```diff
@@ -53,23 +118,61 @@ export default {
...
     },
     openConfirmModal () {
       this.showConfirmModal = true
+    },
+    openEditModal () {
+      this.showEditModal = true
+    },
+    // Stores the incoming prop locally
+    onProp (charge) {
+      this.localCharge = charge
```
Comment (MontgomeryWatts) [inline src/components/ChargeAdmin.vue]: `localCharge` isn't used any where.

File: src/components/ChargeAdmin.vue [hunk 2]
```diff
@@ -44,6 +103,12 @@ export default {
   props: ['charge'],
   data () {
     return {
+      newTitle: null,
```
[the remark this answers is not in the document]
Comment (elijah-parrish) [reply src/components/ChargeAdmin.vue]: Whoops yeah thats dead code. I removed them.

File: src/components/ChargeAdmin.vue [hunk 3]
```diff
@@ -81,6 +184,7 @@ export default {
           that.$router.push({path: '/committee/' + that.charge.committee})
         }, 2000)
       } else if (data.error) {
+        console.log(data.error)
```
[the remark this answers is not in the document]
Comment (elijah-parrish) [reply src/components/ChargeAdmin.vue]: Nope, good catch

File: src/components/ChargeAdmin.vue [hunk 4]
```diff
@@ -53,23 +118,61 @@ export default {
...
+    },
+    openEditModal () {
+      this.showEditModal = true
+    },
+    // Stores the incoming prop locally
+    onProp (charge) {
+      this.localCharge = charge
+    },
+    // Dynamically updates the new prop as the user types
+    updateProp (field, event) {
```
[the remark this answers is not in the document]
Comment (elijah-parrish) [reply src/components/ChargeAdmin.vue]: Thank you! I had researched like mad but that resource finally helped me with what i was missing, the "deep watcher" did the trick.

[the remark this answers is not in the document]
Comment (elijah-parrish) [reply src/components/ChargeAdmin.vue]: The only problem is that with this approach i cannot use the :value="..." in the text inputs, Vue forces me to use :placeholder instead. I feel like populating those fields with text makes more sense for editing a charge, maybe worth some less than best practice code? I'll try to solve with the new method though

[the remark this answers is not in the document]
Comment (lukasyelle) [reply src/components/ChargeAdmin.vue]: This looks good to me now, it seems like you have gotten past that issue right?

[the remark this answers is not in the document]
Comment (elijah-parrish) [reply src/components/ChargeAdmin.vue]: Yes it all works as intended now

File: src/components/ChargeAdmin.vue (hunk 1 above)
Comment (elijah-parrish) [reply src/components/ChargeAdmin.vue]: Good call thats unneeded.

**Pull request merged**